stock_lot_suffix/models/stock_picking.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_channel_category(self):
+        move_lines = self.env["stock.move.line"].search([("picking_id", "=", self.id)])
+        lot_ids = move_lines.mapped("lot_id").ids
+        if lot_ids:
+            lots = self.env["stock.lot"].browse(lot_ids)
```

Comment:
`self` here is a recordset, therefore you need to loop over it.

```suggestion
        for pick in self:
            lots = self.move_line_ids.lot_id
            if lots:
```